core/src/main/java/org/apache/iceberg/SnapshotProducer.java

```diff
@@ -116,6 +117,38 @@ public ThisT scanManifestsWith(ExecutorService executorService) {
     return self();
   }
 
+  @Override
+  public ThisT toBranch(String branch) {
+    throw new UnsupportedOperationException("Performing operations on a branch is currently not supported");
+  }
+
+  /***
+   * Will be used by snapshot producer operations to create a new ref if an invalid branch is passed
```

Comment:
Can we also update the comment, instead of "invalid branch" I think it should be "create a new branch if it does not exist"